dali/pipeline/operators/reader/loader/coco_loader.cc

```diff
@@ -356,6 +346,27 @@ void ParseAnnotationFilesHelper(std::vector<std::string> &annotations_filename,
           image_id_pairs.push_back(std::make_pair(image_file_name, id));
           image_id_to_wh.insert(std::make_pair(id, std::make_pair(width, height)));
         }
+      } else if (0 == strcmp(key, "categories")) {
+        RAPIDJSON_ASSERT(r.PeekType() == kArrayType);
+        r.EnterArray();
+        int id;
+        while (r.NextArrayValue()) {
+          if (r.PeekType() != kObjectType) {
+            continue;
+          }
+          id = -1;
+          r.EnterObject();
+          while (const char* internal_key = r.NextObjectKey()) {
+            if (0 == strcmp(internal_key, "id")) {
+              id = r.GetInt();
```

Comment:
just wondering, is there any way to stop looping through object keys once we have found the `id`?

Reply:
I think `SkipObject()` should do.